Added addSerializer and addDeserializer extensions to SimpleModule for registering KClasses

Neuman968 opened the pull request:
Added extension methods for adding custom serializers and deserializers for a Kotlin KClass that adds both the java class and java type as bound classes. 

This allows for registering serializers like 

```kotlin
    val objectMapper = jacksonObjectMapper()
            .registerModule(SimpleModule()
                    .addSerializer(Double::class, RoundingSerializer())
                    .addDeserializer(Double::class, RoundingDeserializer()))
``` 

Instead of 

```kotlin
    val objectMapper = jacksonObjectMapper()
            .registerModule(SimpleModule()
                    .addSerializer(Double::class.javaObjectType, RoundingSerializer())
                    .addSerializer(Double::class.java, RoundingSerializer())
                    .addDeserializer(Double::class.javaObjectType, RoundingDeserializer())
                    .addDeserializer(Double::class.java, RoundingDeserializer()))
```

Hopefully this can be useful. Thank you for your time and for the awesome project!

dinomite commented:
Great improvement!  Would you add your info to the `release-notes/VERSION-2.x` and `release-notes/CREDITS-2.x`?  I'll get this merged to master and it will go out as part of 2.12.

dinomite commented:
Ack—I've done it a bit wrong.   Just realized that master is tracking Jackson 3.0.0, so I'll fix up the commits I made to put this back on the 2.x tree…

Neuman968 commented:
@dinomite Thank you for your review! I tried to follow the targeted branches from other PR's out from this project. I originally target it for master but wasn't sure so I changed it to 2.12. I just updated the PR to update release-notes/VERSION-2.x and release-notes/CREDITS-2.x Let me know if there is anything else, glad I can help!

dinomite commented:
You did it right, it was my own confusion—I'm still learning the ropes on this project.

dinomite commented:
@Neuman968 I forgot to have you sign the CLA—would you fill this out and email a scan/photo of the result to info at fasterxml dot com?

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf
